book/templates.rst

```diff
@@ -1,200 +1,721 @@
+these values and render them according to the desired design.
+
+Each page template is defined by two files:
+
+* an **XML file** that contains the page structure
+* a **Twig file** that contains the HTML code
 
-To add a new template to your Sulu installation, you just create a new xml file
-in `app/Resources/pages` named by a unique template key. See the following
-file for an example of a template definition:
+For example, the default template -- named "default" -- is defined by the files
+``app/Resources/templates/pages/default.xml`` and
+``app/Resources/views/templates/default.html.twig``. The Sulu Minimal Edition
+also contains a second template named "homepage", which you can find in the
+same directories.
+
+This guide focuses on the configuration of the page structure in the XML file.
+If you want to learn more about rendering the pages in Twig, read
+:doc:`twig`.
+
+Choosing the Template of a Page
+-------------------------------
+
+The template of a page can be selected in the admin interface:
+
+.. figure:: ../img/templates-selection.png
+
+.. Caution::
+
+    A template is shown in the dropdown only if both the XML and the Twig file
+    exist! If it is not, double-check the directories
+    ``app/Resources/templates/pages`` and ``app/Resources/views/templates``.
+
+The name displayed in the dropdown is configured in the ``<meta>`` section of
+the XML:
 
 .. code-block:: xml
 
+    <!-- app/Resources/templates/pages/default.xml -->
     <?xml version="1.0" ?>
     <template xmlns="http://schemas.sulu.io/template/template"
               xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
               xsi:schemaLocation="http://schemas.sulu.io/template/template http://schemas.sulu.io/template/template-1.0.xsd">
-        <key>default</key>
-
-        <view>ClientWebsiteBundle:templates:default</view>
-        <controller>SuluWebsiteBundle:Default:index</controller>
-        <cacheLifetime>2400</cacheLifetime>
+        <!-- ... -->
 
         <meta>
             <title lang="en">Default</title>
         </meta>
 
-        <properties>
-            <property name="title" type="text_line" mandatory="true">
-                <meta>
-                    <title lang="en">Title</title>
-                </meta>
+        <!-- ... -->
+    </template>
 
-                <tag name="sulu.rlp.part"/>
-            </property>
+You can customize the text by changing this property.
 
-            <property name="url" type="resource_locator" mandatory="true">
-                <meta>
-                    <title lang="en">Resourcelocator</title>
-                </meta>
+Creating a Custom Template
+--------------------------
 
-                <tag name="sulu.rlp"/>
-            </property>
+In your projects, you will need several templates for different parts of your
+website. The easiest way is to copy and adjust the default template.
+
+The first thing you need to adjust is the ``<key>``. This is the unique
```

Comment:
this has to be the same as the filename. it is currently a bug but should be mentioned here.